src/com/owncloud/android/files/services/FileDownloader.java

```diff
@@ -441,6 +452,39 @@ private void downloadFile(String downloadKey) {
                                     mCurrentDownload.getRemotePath()
                             );
 
+                    if (!downloadResult.isSuccess() && downloadResult.getException() != null) {
+
+                        // Check network availabality
+                        if (!ConnectivityUtils.isNetworkActive(getApplicationContext())) {
+
+                            // Schedule future retry of failed download due to network error from Android 5
+                            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+
+                                ComponentName mServiceComponent = new ComponentName(this,
+
+                                        RetryDownloadJobService.class);
+
+                                JobInfo.Builder builder;
+                                
+                                builder = new JobInfo.Builder(jobId, mServiceComponent);
+                                // require unmetered network
+                                builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY);
+
+                                // Extra data
+                                PersistableBundle extras = new PersistableBundle();
+                                extras.putString(KEY_FILE_REMOTE_PATH, mCurrentDownload.getFile()
+                                        .getRemotePath());
+
+                                extras.putString(KEY_ACCOUNT_NAME, mCurrentAccount.name);
+
+                                builder.setExtras(extras);
+                                JobScheduler jobScheduler = (JobScheduler) getApplicationContext().
```

Comment:
Use ```this```  instead of ```getApplicationContext()```. ```FileDownloader``` is a ```Context``` , and we only need a context here to access a system service, not to bind anything to it, so the current context is good enough.